src/validators/tuple.ts

```diff
@@ -26,23 +29,31 @@ export const tuple = <U extends readonly (Primitives | SomeSchema<any>)[]>(
 
         if (isSchema(valueOrSchema)) {
           const r = valueOrSchema.__validate(value);
-          result[i] = r.value as unknown;
-          isError ||= r._t === 'left';
-          continue;
-        } else {
-          if (valueOrSchema === value) {
-            result[i] = value;
-            continue;
+          if (r._t === 'right' || r._t === 'nextValid') {
+            result[i] = r.value as unknown;
           } else {
-            result[i] = new ValidationError(this, validatorsOrLiterals);
             isError = true;
-            continue;
+            result[i] = new ValidationError(valueOrSchema, value, r);
           }
+        } else if (valueOrSchema === value) {
+          result[i] = value;
+        } else {
+          // TODO: question: create constant/literal validator?
```

Comment:
@mmiszy WDYT?

Reply:
oneOf can be used for this purpose

Reply:
Ok, so my proposition is to stop supporting literals in a tuple and accept only schema objects as arguments.

Reply:
Tuple now accepts only schema objects